### PR TITLE
Quota view

### PR DESCRIPTION
**Pull request by a-belhadj**
Need to fix overlap in table header before approving it
image

### CODE REVIEW

File: resource_tracker_v2/urls.py [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
+    path('attributes/<int:attribute_definition_id>/', views.AttributeDefinitionDetailView.as_view(),
+         name='attributedefinition_detail'),
     path('attributes/<int:attribute_definition_id>/edit/', views.AttributeDefinitionEditView.as_view(),
          name='attributedefinition_edit'),
     path('attributes/<int:attribute_definition_id>/delete/', views.AttributeDefinitionDeleteView.as_view(),
          name='attributedefinition_delete'),
 
     # resource group
     path('resource_group/', views.ResourceGroupListView.as_view(), name='resourcegroup_list'),
+    path('resource_group/table', views.ResourceGroupListViewCSV.as_view(), name='resourcegroup_list_table'),
```
Comment (Sispheor) [inline resource_tracker_v2/urls.py:20]: missing last slash

File: profiles/views/quota.py [hunk 2]
```diff
@@ -2,13 +2,22 @@
...
+from profiles.tables.quota_table import QuotaTable
 from profiles.tables.team_quota_limit_table import TeamQuotaLimitTable
 from resource_tracker_v2.models import ResourceAttribute
 
 
+class QuotaListView(SquestListView):
+    model = Quota
+    filterset_class = QuotaFilter
+    table_class = QuotaTable
+    ordering = 'name'
```
Comment (Sispheor) [inline profiles/views/quota.py:18]: image
useless add button

**Comment by Sispheor** [comment]
imagewith dark mode we don't see the left column.
No prod in light mode
image

**Comment by Sispheor** [comment]
the attribute column is not needed here as it will be always the same
image

**Comment by Sispheor** [comment]
image
Limit should become a button that redirect to the edit quota page

File: resource_tracker_v2/models/transformer.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ class Meta:
                                              related_name="consume_from_resource_group")
     consume_from_attribute_definition = ForeignKey(AttributeDefinition, null=True, blank=True, on_delete=SET_NULL,
                                                    related_name="consume_from_attribute_definition")
-    factor = FloatField(null=True, blank=True)
+    factor = FloatField(default=1)
```
Comment (Sispheor) [inline resource_tracker_v2/models/transformer.py:32]: I cannot retrieve why it was not set like this in first place. But there is a reason !

File: templates/base.html [hunk 4]
```diff
@@ -71,7 +70,12 @@
                             </div>
                             <div class="col-sm-6">
                                 <div class="float-right">
-                                    {% block header_button %}{% endblock %}
+                                    {% if extra_html_button_path %}
```
Comment (Sispheor) [inline templates/base.html:73]: what is the difference with `html_button_path` already used in other parts? Do we need both?

Comment (Sispheor) [reply templates/base.html:73]: TODO: remove this block from generic edit
```
{% block header_button %}
    {% if html_button_path %}
        <div class="float-right">
            {% include html_button_path %}
        </div>
    {% endif %}
{% endblock %}
```

And use only extra_html_button_path. Rename html_button_path --> extra_html_button_path

File: project-static/squest/css/squest-dark.css [hunk 5]
```diff
@@ -3,3 +3,6 @@
     background-color: #343a40 !important;
     color: #fff !important;
 }
+.dtfc-fixed-left{
```
Comment (Sispheor) [inline project-static/squest/css/squest-dark.css:6]: Actually removing this make the column working fine
image

Comment (Sispheor) [reply project-static/squest/css/squest-dark.css:6]: Nevermind, it was my cache

File: resource_tracker_v2/models/transformer.py (hunk 3 above)
Comment (a-belhadj) [reply resource_tracker_v2/models/transformer.py:32]: All tests passed

File: resource_tracker_v2/views/attribute_definition_views.py [hunk 6]
```diff
@@ -1,10 +1,36 @@
...
 
+class AttributeDefinitionDetailView(SquestDetailView):
+    model = AttributeDefinition
+    pk_url_kwarg = "attribute_definition_id"
+    template_name = "resource_tracking_v2/attributedefinition_detail.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['quotas'] = QuotaTable(
+            Quota.get_queryset_for_user(self.request.user, "profiles.view_quota") & self.object.quotas.distinct(),
```
Comment (Sispheor) [inline resource_tracker_v2/views/attribute_definition_views.py:18]: check perm

File: templates/profiles/globalpermission_detail.html [hunk 7]
```diff
@@ -35,7 +36,11 @@
                     <div class="card-body">
                         <div class="tab-content">
                             <div class="tab-pane active" id="users">
-                                {% render_table users %}
+                                {% if users %}
+                                    {% render_table users %}
+                                {% else %}
+                                    <p> Permission <b>profiles.view_users_globalpermission</b> required</p>
+                                {% endif %}
```
Comment (Sispheor) [inline templates/profiles/globalpermission_detail.html:43]: the end user should not know that he is missing some info due to the perm

**Review by Sispheor — APPROVED** [review]

**Pull request merged**